server/core/terraform/terraform_client.go

```diff
@@ -262,6 +280,90 @@ func (c *DefaultClient) TerraformBinDir() string {
 	return c.binDir
 }
 
+// ListAvailableVersions returns all available version of Terraform. If downloads are not allowed, this will return an empty list.
+func (c *DefaultClient) ListAvailableVersions(log logging.SimpleLogging) ([]string, error) {
+	url := fmt.Sprintf("%s/terraform", c.downloadBaseURL)
+
+	if !c.downloadAllowed {
+		log.Debug("Terraform downloads disabled. Won't list Terraform versions available at %s", url)
+		return []string{}, nil
+	}
+
+	log.Debug("Listing Terraform versions available at: %s", url)
+	// terraform-switcher calls os.Exit(1) if it fails to successfully GET the configured URL.
+	// So, before calling it, test if we can connect. Then we can return an error instead if the request fails.
+	resp, err := http.Get(url) // #nosec G107 -- terraform-switch makes this same call below. Also, we don't process the response payload.
+	if err != nil || resp.StatusCode != 200 {
+		return nil, fmt.Errorf("Unable to list Terraform versions: %s", err)
+	}
```

Comment:
http client requires body close not to leak memory.

```suggestion
	resp, err := http.Get(url)
	if err != nil {
		return nil, fmt.Errorf("Unable to list Terraform versions: %s", err)
	}
	defer resp.Body.Close() // nolint: errcheck
	if resp.StatusCode != http.StatusOK {
		return nil, fmt.Errorf("list terravorm version unexpected status code: %d, url: %s", resp.StatusCode, url)
	}
```

BTW, generally we should not use http.Get, because it uses default client.  terraform.DefaultClient can have httpClient which can be mocked. But currently atlantis use DefaultClient by default, so no problem on this PR.
https://github.com/runatlantis/atlantis/blob/9a38e354936fbf93282e4e934fa3439fc3bfd102/server/server.go#L262

Reply:
Addressed in https://github.com/runatlantis/atlantis/pull/2843/commits/a21732f570273e016c0af476b69bb74fe1c91302.